spec/requests/user_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe User, :type => :request do
+  let(:user) { create(:user, :admin) }
+  let(:headers) { { "Authorization": sign_in(user) } }
+  let(:target_user) { create(:user) }
+
+  it "creates a User" do
```

Comment:
Acho que aqui, fica top se vc separar cada verbo por `describes`

```ruby
describe 'POST users_path' do
  subject { post users_path, params: { user: attributes_for(:user) }, headers: headers }

  it 'returns http_status created' do
    subject

    expect(response).to have_http_status(:created)
  end

  it 'creates an User' do
    expect{ subject }.to change(User, :count).by(1)
  ed
end

```

Reply:
Eu acho que concordo com a organização, mas não faz muito sentido ter um `it` só pra testar o `http_status` retornado, dado que eu só posso ter 201 se um user for criado e um 400 se a requisição for mal feita...

Reply:
Blz, até concordo, são só 2 caminhos pra requisição, mas eu acho importante testar a parte onde cria ou nao um usuário. Por exemplo se alguem faz uma requisição colocando um CPF abobrinha, a gente sabe que tem as validations la, mas o seu teste tem que garantir.